core/emf-api/src/main/java/org/eclipse/sensinact/core/emf/dto/EMFGenericDto.java

```diff
@@ -29,6 +29,14 @@ public final class EMFGenericDto extends BaseValueDto {
     /** The {@link EClass} of the provider */
     public EClass modelEClass;
 
+    /**
+     * The name of the {@link EClass} of a service. This must be set, when updating
+     * a {@link DynamicProvider}, where a service might not be available yet, it is
+     * an alternative to the serviceEClass and will be overwritten by the
+     * serviceEClass is set..
+     */
+    public String serviceEClassName;
+
```

Comment:
We should also add this in the `BaseValueDto` with the `@ServiceModel` annotation on it